File: 01.JS-Basic/WHILE-ЦИКЪЛ Упражнение/Cake.js
function cake(input) {

    let cakeSize = Number(input[0]) * Number(input[1])
    let index = 2

    while (cakeSize > 0 && input[index] !== "STOP") {
        let piecesTaken = input[index]
        cakeSize -= piecesTaken
        index++
    }
    if (cakeSize > 0) {
        console.log(`${cakeSize} pieces are left.`)
    } else {
        console.log(`No more cake left! You need ${Math.abs(cakeSize)} pieces more.`)
    }
}
cake(["10",
    "2",
    "2",
    "4",
    "6",
    "STOP"])

File: JAVA-Basic/JavaBasic/src/WhileLoop/Exercise/Cake.java
package WhileLoop.Exercise;

import java.util.Scanner;

public class Cake {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int w = Integer.parseInt(scan.nextLine());
        int l = Integer.parseInt(scan.nextLine());

        int cakeSize = w * l;

        String command = scan.nextLine();

        while (!command.equals("STOP")) {
            int currentPieces = Integer.parseInt(command);
            cakeSize -= currentPieces;
            if (cakeSize < 0) {
                System.out.printf("No more cake left! You need %d pieces more.", Math.abs(cakeSize));
                return;
            }
            command = scan.nextLine();
        }
        System.out.printf("%d pieces are left.", cakeSize);
    }
}
